contracts/test/StakeTable.t.sol

```diff
@@ -633,6 +652,596 @@ contract StakeTable_register_Test is LightClientCommonTest {
+        emit S.ValidatorExit(validator);
+        stakeTable.deregisterValidator();
+        vm.stopPrank();
+
+        // Test 4: Cannot exit again after already exited
+        vm.startPrank(validator);
+        vm.expectRevert(S.ValidatorAlreadyExited.selector);
+        stakeTable.deregisterValidator();
+        vm.stopPrank();
+
+        // Verify validator status after exit
+        (uint256 validatorAmountDelegated, S.ValidatorStatus status) =
+            stakeTable.validators(validator);
+        assertEq(validatorAmountDelegated, 0);
+        assertEq(uint256(status), uint256(S.ValidatorStatus.Exited));
+    }
+
+    function test_ConsensusKeyUpdateAuthorization() public {
+        // Setup - register initial validator
+        (
+            BN254.G2Point memory blsVK,
+            EdOnBN254.EdOnBN254Point memory schnorrVK,
+            BN254.G1Point memory sig
+        ) = genClientWallet(validator, seed1);
+
+        vm.startPrank(validator);
+        stakeTable.registerValidator(blsVK, schnorrVK, sig, COMMISSION);
+        vm.stopPrank();
+
+        (
+            BN254.G2Point memory newBlsVK,
+            EdOnBN254.EdOnBN254Point memory newSchnorrVK,
+            BN254.G1Point memory newSig
+        ) = genClientWallet(validator, seed2);
+
+        // Test 1: Non-validator cannot update keys
+        address nonValidator = makeAddr("nonValidator");
+        vm.startPrank(nonValidator);
+        vm.expectRevert(S.ValidatorInactive.selector);
+        stakeTable.updateConsensusKeys(newBlsVK, newSchnorrVK, newSig);
+        vm.stopPrank();
+
+        // Test 2: Successful update by  validator
+        vm.startPrank(validator);
+        vm.expectEmit(false, false, false, true, address(stakeTable));
+        emit S.ConsensusKeysUpdated(validator, newBlsVK, newSchnorrVK);
+        stakeTable.updateConsensusKeys(newBlsVK, newSchnorrVK, newSig);
+        (uint256 validatorAmountDelegated, S.ValidatorStatus status) =
+            stakeTable.validators(validator);
+        assertEq(validatorAmountDelegated, 0);
+        assertEq(uint256(status), uint256(S.ValidatorStatus.Active));
+
+        // Test 3: Cannot update with same BLS key
+        vm.expectRevert(S.BlsKeyAlreadyUsed.selector);
+        stakeTable.updateConsensusKeys(newBlsVK, newSchnorrVK, newSig);
+        vm.stopPrank();
+
+        // Test 4: Cannot update after exit
+        vm.startPrank(validator);
+        stakeTable.deregisterValidator();
+
+        // Generate another set of new keys
+        (
+            BN254.G2Point memory postExitBlsVK,
+            EdOnBN254.EdOnBN254Point memory postExitSchnorrVK,
+            BN254.G1Point memory postExitSig
+        ) = genClientWallet(validator, "145");
+
+        vm.expectRevert(S.ValidatorAlreadyExited.selector);
+        stakeTable.updateConsensusKeys(postExitBlsVK, postExitSchnorrVK, postExitSig);
+        vm.stopPrank();
+    }
+
+    function test_ValidatorSelfDelegation() public {
```

Comment:
i saw this acceptance criteria in the [book](https://github.com/EspressoSystems/book/blob/main/pages/stake-table-contract-requirements.tex#L142), but i wonder why we want this behaviour @sveitser

Reply:
The previous design separated out solo stakers. This is to explicitly make it clear that the use-case of delegating to one's own node is covered by the new design.